src/fastcdc.rs

```diff
@@ -0,0 +1,281 @@
+    }
+}
+
+impl FastCDC {
+    /// Create new FastCDC engine with default chunking settings
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Create new `FastCDC` engine with custom chunking settings
+    ///
+    /// `chunk_bits` is number of bits that need to match in
+    /// the edge condition. `CHUNK_BITS` constant is the default.
+    pub fn new_with_chunk_bits(chunk_bits: u32) -> Self {
+        Self {
+            current_chunk_size: 0,
+            gear: Gear::new_with_chunk_bits(chunk_bits),
+        }
+    }
+
+    /// Find chunk edge using `FastCDC` defaults.
+    ///
+    /// See `Engine::find_chunk_edge_cond`.
+    pub fn find_chunk_edge(&mut self, mut buf: &[u8]) -> Option<(usize, u64)> {
+
+        const DIGEST_SIZE: usize = 64;
+        debug_assert_eq!(
+            mem::size_of::<<Self as Engine>::Digest>() * 8,
+            DIGEST_SIZE
+            );
+
+        const SPREAD_BITS: u32 = 3;
+        const WINDOW_SIZE: usize = 64;
+
+        let min_shift = DIGEST_SIZE as u32 - self.gear.chunk_bits - SPREAD_BITS;
+        let max_shift = DIGEST_SIZE as u32 - self.gear.chunk_bits + SPREAD_BITS;
+        let min_size = (1 << (self.gear.chunk_bits - SPREAD_BITS)) as u64;
+        let ignore_size = min_size - WINDOW_SIZE as u64;
+        let avg_size = (1 << self.gear.chunk_bits) as u64;
+        let max_size = (1 << (self.gear.chunk_bits + SPREAD_BITS)) as u64;
+
+        let mut cur_offset = 0usize;
+
+        loop {
+            debug_assert!(self.current_chunk_size < max_size);
+            debug_assert!(cur_offset < max_size as usize);
+
+            if buf.is_empty() {
+                return None
+            }
+
+            // ignore bytes that are not going to influence the digest
+            if self.current_chunk_size < ignore_size {
+                let skip_bytes = cmp::min(ignore_size  - self.current_chunk_size, buf.len() as u64);
+                self.current_chunk_size += skip_bytes;
+                cur_offset += skip_bytes as usize;
+                buf = &buf[skip_bytes as usize..];
+            }
+
+            // ignore edges in bytes that are smaller than min_size
+            if self.current_chunk_size < min_size {
+                let roll_bytes = cmp::min(min_size - self.current_chunk_size,
+                                          buf.len() as u64);
+                self.gear.roll(&buf[..roll_bytes as usize]);
+                self.current_chunk_size += roll_bytes;
+                cur_offset += roll_bytes as usize;
+                buf = &buf[roll_bytes as usize..];
+            }
+
+            // roll through early bytes with smaller probability
+            if self.current_chunk_size < avg_size {
+                let roll_bytes = cmp::min(avg_size - self.current_chunk_size,
+                                          buf.len() as u64);
+                let result = self.gear.find_chunk_edge_cond(buf, |e: &Gear| (e.digest() >> min_shift) == 0);
```

Comment:
Hm, you're quite right. I've misread this; the window here is essentially the width of the digest; 64 bytes.

The authors of the paper describe the mask they use in the algorithm as being empirically derived, but infuriatingly not giving any details about it. You'd think that taking the largest window would be best, but apparently not? Apparently it works best when the contributing bits are split approximately uniformly across the 64bit digest?